src/status_im2/contexts/wallet/events.cljs

```diff
@@ -311,6 +312,42 @@
                :wallet/valid-ens-or-address?
                false)}))
 
+(rf/reg-event-fx :wallet/find-ens
+ (fn [{:keys [db]} [input cb]]
+   (let [contacts (rf/sub [:contacts/active])
+         result   (if (empty? input)
+                    []
+                    (filter #(string/starts-with? (or (:ens-name %) "") input) contacts))]
+     (if (and input (empty? result))
+       (rf/dispatch [:wallet/search-ens input cb])
+       {:db (assoc db
+                   :wallet/local-suggestions
+                   (map #(assoc % :type item-types/saved-address) result)
+                   :wallet/valid-ens-or-address? (not-empty result))}))))
+
+(rf/reg-event-fx :wallet/search-ens
+ (fn [_ [input cb]]
+   (let [chain-id (rf/sub [:chain-id])
```

Comment:
We should not use subscriptions inside event handlers. It could lead to memory leaks.
https://day8.github.io/re-frame/FAQs/UseASubscriptionInAnEventHandler/

We should be using helper functions to retrieve the data